pages/articles/authentication.md

```diff
@@ -155,3 +155,44 @@ Permission | Permission type | Id | Type
 Connect-PnPOnline [yourtenant].sharepoint.com -Interactive -ClientId [clientid] -Tenant [yourtenant].onmicrosoft.com -AzureEnvironment [USGovernment|USGovernmentHigh|USGovernmentDoD|Germany|China]
 ```
 The AzureEnvironment parameter only allows one value. Select the correct one that matches your cloud deployment.
+
+## Silent Authentication with Credentials for running in Pipelines
+
+For running `Connect-PnPOnline` with user credentials in Azure DevOps pipeline, you need to make sure that authentication in your Azure AD application is configured to allow public client. 
+
+Public client can be configured from the Azure portal from the Authentication Blade in the application or by setting the `allowPublicClient` property in the application's manifest to true.
+
+`username` and `password` for service account can be stored as secret pipeline variables and can be referenced in the script to achieve complete automation.
+
+## Silent Authentication with Credentials and MFA for running in Azure DevOps Pipelines with Microsoft Hosted Agents
+### Identify the possible IP ranges for Microsoft-hosted agents
+
+- Identify the [region for your organization](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/change-organization-location?view=azure-devops) in Organization settings.
+- Identify the [Azure Geography](https://azure.microsoft.com/global-infrastructure/geographies/) for your organization's region.
+- Map the names of the regions in your geography to the format used in the weekly file, following the format of AzureCloud., such as AzureCloud.westus. You can map the names of the regions from the Azure Geography list to the format used in the weekly file.
+- You can find the weekly IP file at the following URL: https://www.microsoft.com/en-us/download/details.aspx?id=56519
+
+For example, if your organization is located in the South East Asia region, you would map it to the format AzureCloud.SouthEastAsia.
+
+### Create a named location in Azure AD conditional access
+
+- Go to Azure AD conditional access
+- Open named location blade, click on `+ IP Ranges Location`
+- Enter the IP ranges for Microsoft Hosted Agents, `Mark as trusted location` should be checked.
```

Comment:
Can you also please add a screenshot or a gif for this ?

Reply:
Done. Thanks for the review.